tikv-client/src/main/java/com/pingcap/tikv/PDClient.java

```diff
@@ -347,22 +347,33 @@ LeaderWrapper getLeaderWrapper() {
     return leaderWrapper;
   }
 
-  private GetMembersResponse getMembers(URI url) {
-    try {
-      ManagedChannel probChan = channelFactory.getChannel(url.getHost() + ":" + url.getPort());
-      PDGrpc.PDBlockingStub stub = PDGrpc.newBlockingStub(probChan);
-      GetMembersRequest request =
-          GetMembersRequest.newBuilder().setHeader(RequestHeader.getDefaultInstance()).build();
-      GetMembersResponse resp = stub.getMembers(request);
-      // check if the response contains a valid leader
-      if (resp != null && resp.getLeader().getMemberId() == 0) {
-        return null;
+  private GetMembersResponse doGetMembers(BackOffer backOffer, URI url) {
+    while (true) {
+      try {
+        ManagedChannel probChan = channelFactory.getChannel(url.getHost() + ":" + url.getPort());
+        PDGrpc.PDBlockingStub stub = PDGrpc.newBlockingStub(probChan);
+        GetMembersRequest request =
+            GetMembersRequest.newBuilder().setHeader(RequestHeader.getDefaultInstance()).build();
+        GetMembersResponse resp = stub.getMembers(request);
+        // check if the response contains a valid leader
+        if (resp != null && resp.getLeader().getMemberId() == 0) {
+          return null;
+        }
+        return resp;
+      } catch (Exception e) {
+        logger.warn("failed to get member from pd server.", e);
+        backOffer.doBackOff(BackOffFuncType.BoPDRPC, e);
       }
-      return resp;
+    }
+  }
+
+  private GetMembersResponse getMembers(URI uri) {
+    BackOffer backOffer = ConcreteBackOffer.newCustomBackOff(BackOffer.PD_INFO_BACKOFF);
+    try {
+      return doGetMembers(backOffer, uri);
     } catch (Exception e) {
-      logger.warn("failed to get member from pd server.", e);
```

Comment:
Why swallow the exception

Reply:
Shall we print an exception stack here?
Now, tikv-client is `return null` here and null will be checked at the call.

Reply:
Yes